src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -540,6 +540,13 @@ def _report_single_requirement_conflict(self, req, parent):
             req_disp,
             ", ".join(versions) or "none",
         )
+        if str(req) == "requirements.txt":
+            logger.critical(
```

Comment:
Critical is for things critical to pip itself, this shouldn’t be set that high. If you want a red message, `error` is enough, but `warning` or even `info` might be suitable for a hint as well, I don’t have a strong preference.

Reply:
An alternative is to merge this message to the previous one:

```python
message = "Could not find a version that satisfies the requirement %s (from versions: %s)"
if str(req) == "requirements.txt":
    message += "\nHINT: You are atempting to (...snipped)"
logger.critical(message, req_disp, ", ".join(versions) or "none")
```

Reply:
I'm going to go with `info` because there is not really a danger here, so `warning` seems excessive. Also, I don't think modifying the existing message is a good idea because I don't want to break any functionality / tests that rely on that message being what it is.